test/test_sitemaps_from_robotstxt.py

```diff
@@ -0,0 +1,576 @@
+Disallow: /?page=hotellist_json
+Disallow: /reviewpage2/
+
+
+User-agent: Yandex
+Host: nochi.com
+Allow: /
+Disallow: /reviewpage/
+
+Disallow: /ajax/
+Disallow: /?page=stat
+Disallow: /?page=hotel_ajax
+Disallow: /?page=hotellist_json
+Sitemap: http://nochi.com/data/sitemaps/ru_index.xml
+""".encode('utf-8')
+        )
+    ]
+
+    records = [make_robots_txt_record(robots_txt_url, robots_txt_content)
+                for robots_txt_url, robots_txt_content in multi_robots_txt_data]
+    job = SitemapExtractorJob()
+    job.init_accumulators(session=spark)
+    rdd = spark.sparkContext.parallelize(records)
+    _process_jobs_partial = lambda partition_index, records: _process_jobs(partition_index, records, job=job)
+    output = rdd.mapPartitionsWithIndex(_process_jobs_partial)
+    output = output.groupByKey().map(SitemapExtractorJob.reduce_group_by_key_func).collect()
+    assert len(output) == 1
+    assert output[0][0] == 'http://nochi.com/data/sitemaps/ru_index.xml'
+    assert sorted(json.loads(output[0][1])) == sorted(["the-mayflower-hotel-autograph-collection-washington.ibooked.com.br","the-rockies-condominiums-steamboat-springs.booked.net","hotel-flora-venice.booked.kr"])
+    assert job.sitemap_urls_found.value == 3
+    assert job.sitemap_url_invalid.value == 0
+    assert job.sitemap_url_invalid_encoding.value == 0
+    assert job.robots_txt_invalid_url.value == 0
+    assert job.robots_txt_announcing_sitemap.value == 3
+    assert job.robots_txt_with_more_than_50_sitemaps.value == 0
+    assert job.robots_txt_processed.value == 3
+
+
+def test_wrong_encoding_utf16_record(spark):
+    record = make_robots_txt_record("http://ajedrezhoygol.blogspot.com.ar/robots.txt",
+                                    """User-agent: Mediapartners-Google
+Disallow: 
+
+User-agent: *
+Disallow: /search
+Allow: /
+
+Sitemap: http://ajedrezhoygol.blogspot.com/sitemap.xml
+""".encode('utf-16'))
+    job = SitemapExtractorJob()
+    job.init_accumulators(session=spark)
+    results = list(job.process_record(record))
+    assert len(results) == 0
+    assert job.sitemap_urls_found.value == 0
+    assert job.sitemap_url_invalid.value == 0
+    assert job.sitemap_url_invalid_encoding.value == 0
+    assert job.robots_txt_invalid_url.value == 0
+    assert job.robots_txt_announcing_sitemap.value == 0
+    assert job.robots_txt_with_more_than_50_sitemaps.value == 0
+    assert job.robots_txt_processed.value == 1
+
+
+def test_robots_txt_invalid_url_unparseable_netloc(spark):
+    """ Test malformed WARC-Target-URI """
+    record = make_robots_txt_record(
+        warc_target_uri='http://[malformed::url]/robots.txt',
+        response_bytes="""User-agent: *
+Disallow: /admin/
+
+Sitemap: http://example.com/sitemap.xml
+Sitemap: http://example.com/sitemap2.xml
+""".encode('utf-8')
+    )
+
+    job = SitemapExtractorJob()
+    job.init_accumulators(session=spark)
+    results = list(job.process_record(record))
+    assert len(results) == 0
```

Comment:
Shouldn't it be 2 because both sitemap URLs are valid? If the WARC-Target-URI does not validate, it only effects the cross-submit list. But this will hardly happen. In all crawls since Aug 2018 the WARC-Target-URI is validated as URI.